### ISSUE TITLE
Correct XML encoding for OCS

### ISSUE THREAD

**Issue by nchizhov** [glpi-project/glpi-agent#434]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

If using OCS-client GLPI-Agent not correct send xml if in it cyrillic text.

### To reproduce

Incorrect cyrillyc text in OCS

### Expected behavior

For fix it:
1. Open file: lib/GLPI/Agent/HTTP/Client/OCS.pm
2. Find string `$request_content = $self->compress(encode('UTF-8', $request_content));`
3. Change it to: `$request_content = $self->compress($request_content)`;

### Operating system

Windows

### GLPI Agent version

Nightly build (git version in additional context below)

### GLPI version

Other (See additional context below)

### GLPIInventory plugin or FusionInventory for GLPI plugin version

Other (See additional context below)

### Additional context

Using OCS

**Comment by g-bougard** [comment]
Hi @nchizhov 

this is the first time I hear somebody he uses glpi-agent with OCS. Of course you should know OCS provides a dedicated client. Did you try it ? If yes, can you tell us why choosing glpi-agent then ?

As you didn't tell us which nightly version you're using, be sure to use at least the latest 1.5 agent which was released last week. This new version includes a XML support refactoring which still can fix your issue, but also few fixes related to string lookup in registry and their decoding. But I may have missed something.

Actually, the issue still can be a OCS server issue as nobody shared such a problem with GLPI. You'll first have to demonstrate the sent XML is wrongly encoded. Can you download the XML from OCS server if it is stored there ? Of course, we need a XML generated by a not modified glpi-agent. You may also just share the badly encoded XML node as this can tell us there's a bug in another place. Maybe just generating the XML with `glpi-inventory` script can still demonstrate your issue.

**Comment by nchizhov** [comment]
I'm using GLPI-Agent for Inventory for my inventory server program and this program using same protocol as OSC.

**Comment by g-bougard** [comment]
Hi @nchizhov 
if it's your own program, I guess you can store the received file or modify it to store it, so please share it so I can investigate in the right way.

**Comment by nchizhov** [comment]
In attach raw XML-file. 
As you can see - cyrillic letters not show correct, because GLPI-Agent encode utf-8 twice.

Using latest GLPI-Agent night build - 1.6-gitbad7806d
[test-glpi.zip](https://github.com/glpi-project/glpi-agent/files/11903634/test-glpi.zip)

**Comment by g-bougard** [comment]
Thank you for the sharing.
Can you eventually also provide the json version as this can help to identify why there's a double-encoding ?
You can generate it with the following command from the same computer:
```
glpi-inventory --json
```

**Comment by nchizhov** [comment]
Generated JSON-file from same computer attached
[test-glpi.zip](https://github.com/glpi-project/glpi-agent/files/11905130/test-glpi.zip)

**Comment by g-bougard** [comment]
Hi @nchizhov 

thank you, this file seems correctly encoded and so this shows me there's no problem when decoding data during data extraction from the system: the agent seems catch the expected values.

So I made a test using an agent as proxy so I can force the remote agent to always send a XML and not switch to json format which is well-formed. From a french win10, I was able to reproduce the same problem. And so I admit now you're totally right. Indeed for GLPI Agent 1.5, I refactored XML support and now the content is provided earlier encoded in UTF-8. This makes the call to encode as UTF-8 in GLPI::Agent::HTTP::Client::OCS wrong. I missed that problem as GLPI-Agent has a mechanism to switch to json when communicating to GLPI 10, and no body noticed that problem before you. I guess this also can be a problem for people using GLPI-Agent 1.5 with FusionInventory plugin.

So a big thank you for your submission.
